console/console-init/ui/src/Components/Common/DropdownComponent.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from "react";
+import { Dropdown, DropdownToggle, DropdownItem } from "@patternfly/react-core";
+import { FilterIcon } from "@patternfly/react-icons";
+
+export interface IDropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface IDropdown {
+  options: IDropdownOption[];
+  isOpen: boolean;
+  setIsOpen: () => void;
+  value: string;
+  onSelect: (item: any) => void;
+}
+
+export const DropdownComponent: React.FunctionComponent<IDropdown> = ({
```

Comment:
`Component` in the name isn't a great practice. I'm not sure about the goal for this component, what problem is it solving?

Reply:
Dropdown for filter is used in many places and to avoid same code being copied. I created a component for this. Should I remove this component?

Reply:
I don't know. I don't like it the way it is. Maybe it was called `FilterMenu` would be more clear what's this for (since the filter icon is hardcoded in the component, so this isn't really a generic reusable menu). And if this didn't require the `isOpen` and `setIsOpen` props to be passed; why can't that be handled internally by the component itself?